torchgeo/trainers/chesapeake.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for the Chesapeake datasets."""
+
+from typing import Any, Dict, Optional, cast
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import segmentation_models_pytorch as smp
+import torch
+import torch.nn as nn
+from pytorch_lightning.core.datamodule import LightningDataModule
+from pytorch_lightning.core.lightning import LightningModule
+from torch import Tensor
+from torch.nn.modules import Module
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.data import DataLoader
+from torch.utils.tensorboard import SummaryWriter  # type: ignore[attr-defined]
+from torchmetrics import Accuracy, IoU
+
+from ..datasets import Chesapeake7, ChesapeakeCVPR
+from ..samplers import RandomBatchGeoSampler
+
+# https://github.com/pytorch/pytorch/issues/60979
+# https://github.com/pytorch/pytorch/pull/61045
+DataLoader.__module__ = "torch.utils.data"
+Module.__module__ = "torch.nn"
+
+CMAP = matplotlib.colors.ListedColormap(
+    [np.array(Chesapeake7.cmap[i + 1]) / 255.0 for i in range(6)]
+)
+
+
+class ChesapeakeCVPRSegmentationTask(LightningModule):
```

Comment:
These trainer modules desperately need unit tests at some point, our overall code coverage is plummeting, see #109

Reply:
Have you tried running a trainer :)?

We ran into a very obvious problem in "tested" code this morning: I wanted to run `python train.py` with the cyclone dataset. This works in my environment because the dataset is where I expect and this works in tests because we have fake data where we expect it. This did _not_ work in practice because other people don't have the dataset downloaded.

Reply:
At the moment `train.py` isn't "tested" code. In the long run, I want to have integration tests that use real data and may take several hours to run. These will only be run on release branches, not on main/PRs. Everything is set up to do this except someone needs to actually write the tests. The unit tests will only be for coverage and catching minor issues.